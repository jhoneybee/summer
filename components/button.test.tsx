/**
 *  TODO 任务完成进度
 * 
 *  - [x] test block 属性
 *  - [x] test disabled 属性
 *  - [x] test href 属性
 *  - [ ] test type 属性
 *  - [ ] test onClick 事件
 */
import React from 'react';
import { render, screen, fireEvent, act } from '@testing-library/react';

import Button from './button';

test('test block', async () => {
    render(
        <div
            style={{
                width: 200
            }}
        >
            <Button
                block
                role='button'
            >
                测试按钮
            </Button>
        </div>
    );

    expect(await screen.findByRole('button')).toMatchSnapshot();
});

test('test disabled', async () => {
    render(
        <Button
            disabled
            role='button'
        >
            测试按钮
        </Button>
    );
    expect(await screen.findByRole('button')).toMatchSnapshot();
});

test('test href[_self]', async () => {
    render(
        <Button
            href="https://www.google.com"
            role='button'
        >
            测试按钮
        </Button>
    );
    expect(await screen.findByRole('button')).toMatchSnapshot();
});

test('test href[_blank]', async () => {
    render(
        <Button
            href="_blank->https://www.google.com"
            role='button'
        >
            测试按钮
        </Button>
    );
    expect(await screen.findByRole('button')).toMatchSnapshot();
});

test('test href[_parent]', async () => {
    render(
        <Button
            href="_blank->https://www.google.com"
            role='button'
        >
            测试按钮
        </Button>
    );
    expect(await screen.findByRole('button')).toMatchSnapshot();
});

test('test href[_top]', async () => {
    render(
        <Button
            href="_blank->https://www.google.com"
            role='button'
        >
            测试按钮
        </Button>
    );
    expect(await screen.findByRole('button')).toMatchSnapshot();
});

test('test type[primary]', async () => {
    render(
        <Button
            type='primary'
            role='button'
        >
            测试按钮
        </Button>
    );
    expect(await screen.findByRole('button')).toMatchSnapshot();
});

test('test type[text]', async () => {
    render(
        <Button
            type='text'
            role='button'
        >
            测试按钮
        </Button>
    );
    expect(await screen.findByRole('button')).toMatchSnapshot();
});

test('test type[text]', async () => {
    render(
        <Button
            type='text'
            role='button'
        >
            测试按钮
        </Button>
    );
    expect(await screen.findByRole('button')).toMatchSnapshot();
});

test('test danger', async () => {
    render(
        <Button
            danger
            role='button'
        >
            测试按钮
        </Button>
    );
    expect(await screen.findByRole('button')).toMatchSnapshot();
});

test('test danger type[text]', async () => {
    render(
        <Button
            danger
            type='text'
            role='button'
        >
            测试按钮
        </Button>
    );
    expect(await screen.findByRole('button')).toMatchSnapshot();
});


test('test onClick', async () => {
    const OnClickButton = () => {
        const [text, setText] = React.useState('测试按钮')
        return (
            <Button
            danger
            type='text'
            role='button'
            onClick={() => {
                setText('改变的按钮');
            }}
            >
                {text}
            </Button>
        )
        
    }
    const { findByRole } = render(
        <OnClickButton />
    );

    const button = await findByRole('button');
    fireEvent.click(button)
    expect(button).toMatchSnapshot();
});

test('test onClick async', async () => {
    await act(async () => {
        const OnClickButton = () => {
            const [text, setText] = React.useState('测试按钮')
            return (
                <Button
                danger
                type='text'
                role='button'
                onClick={async () => {
                    setText('改变的按钮');
                }}
                >
                    {text}
                </Button>
            )
            
        }
        const { findByRole } = render(
            <OnClickButton />
        );

        const button = await findByRole('button');
        fireEvent.click(button)
        expect(button).toMatchSnapshot();
    })
});

test('test onClick none', async () => {
    const OnClickButton = () => {
        return (
            <Button
            danger
            type='text'
            role='button'
            >
                按钮信息
            </Button>
        )
        
    }
    const { findByRole } = render(
        <OnClickButton />
    );

    const button = await findByRole('button');
    fireEvent.click(button)
    expect(button).toMatchSnapshot();
});


