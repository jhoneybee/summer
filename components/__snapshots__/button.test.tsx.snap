// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test block 1`] = `
<button
  class="sc-bdfBwQ gLkOas"
  role="button"
>
  测试按钮
</button>
`;

exports[`test danger 1`] = `
<button
  class="sc-bdfBwQ jemraX"
  role="button"
>
  测试按钮
</button>
`;

exports[`test danger type[text] 1`] = `
<a
  class="sc-bdfBwQ lgXYmG"
  role="button"
>
  测试按钮
</a>
`;

exports[`test disabled 1`] = `
<button
  class="sc-bdfBwQ iwcedJ"
  disabled=""
  role="button"
>
  测试按钮
</button>
`;

exports[`test href[_blank] 1`] = `
<a
  class="sc-bdfBwQ dlbZId"
  href="https://www.google.com"
  role="button"
  target="_blank"
>
  测试按钮
</a>
`;

exports[`test href[_parent] 1`] = `
<a
  class="sc-bdfBwQ dlbZId"
  href="https://www.google.com"
  role="button"
  target="_blank"
>
  测试按钮
</a>
`;

exports[`test href[_self] 1`] = `
<a
  class="sc-bdfBwQ dlbZId"
  href="https://www.google.com"
  role="button"
  target="_self"
>
  测试按钮
</a>
`;

exports[`test href[_top] 1`] = `
<a
  class="sc-bdfBwQ dlbZId"
  href="https://www.google.com"
  role="button"
  target="_blank"
>
  测试按钮
</a>
`;

exports[`test onClick 1`] = `
<a
  class="sc-bdfBwQ lgXYmG"
  role="button"
>
  改变的按钮
</a>
`;

exports[`test onClick async 1`] = `
<a
  class="sc-bdfBwQ jImbRQ"
  disabled=""
  role="button"
>
  改变的按钮
</a>
`;

exports[`test onClick none 1`] = `
<a
  class="sc-bdfBwQ lgXYmG"
  role="button"
>
  按钮信息
</a>
`;

exports[`test type[primary] 1`] = `
<button
  class="sc-bdfBwQ trsXt"
  role="button"
>
  测试按钮
</button>
`;

exports[`test type[text] 1`] = `
<a
  class="sc-bdfBwQ gKLLHS"
  role="button"
>
  测试按钮
</a>
`;

exports[`test type[text] 2`] = `
<a
  class="sc-bdfBwQ gKLLHS"
  role="button"
>
  测试按钮
</a>
`;
